### PR TITLE
Bring in loopback4-example-rpc-server

### PR DESCRIPTION
**Pull request by bajtos**
The first commit 929737c brings in an exact copy of https://github.com/strongloop/loopback4-example-rpc-server.

The second commit f8d08a5 contains changes necessary to integrate the example with the rest of the monorepo.

See #836

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/example-rpc-server/README.md [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
-Install dependencies and start the app:
-```sh
-npm install
+
+1. Install the new loopback CLI toolkit.
+```
+npm i -g @loopback/cli
+```
+
+2. Download the "getting-started" application.
```
Comment (b-admike) [inline packages/example-rpc-server/README.md]: I'm guessing you're adding this to the `lb4 example` prompt and editing these instructions?

File: packages/example-rpc-server/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@loopback/example-rpc-server",
+  "version": "4.0.0-alpha.0",
+  "description": "A basic RPC server using a made-up protocol.",
+  "keywords": [
+    "loopback-application",
+    "loopback"
+  ],
+  "main": "./dist/src/index.js",
```
Comment (virkt25) [inline packages/example-rpc-server/package.json]: Shouldn't this point to `index.js`?

File: packages/example-rpc-server/test/mocha.opts [hunk 3]
```diff
@@ -0,0 +1 @@
+--recursive
```
Comment (virkt25) [inline packages/example-rpc-server/test/mocha.opts]: This file can be removed as it's not used by the test script in `package.json`

File: packages/example-rpc-server/package.json (hunk 2 above)
Comment (virkt25) [reply packages/example-rpc-server/package.json]: Actually it appears that `index.d.ts` and `index.js` are missing in this example yet it supports Node 6 so I'm actually confused as to how this even passes on 6 rn.

File: packages/example-rpc-server/tslint.build.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (virkt25) [inline packages/example-rpc-server/tslint.build.json]: We don't need to a tslint file when using `lb-tslint`. This file can be removed.

File: packages/example-rpc-server/tslint.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (virkt25) [inline packages/example-rpc-server/tslint.json]: This file can be removed. None of the other `example-*` repos have it.

**Comment by bajtos** [comment]
Thank you @kjdelisle and @virkt25 for your detailed review. So many trivial mistakes, I feel bad about not being more diligent when checking my changes myself.

I believe I have addressed all your comments now.

**Comment by bajtos** [comment]
FWIW, the build is failing because of a know problem in one of our dependencies, see https://github.com/sinonjs/nise/issues/27 and https://github.com/sinonjs/nise/pull/29

**Pull request merged**